benchmarks/__main__.py

```diff
@@ -1,27 +1,60 @@
 """Executes the benchmarking module."""
 # Run this via 'python -m benchmarks' from the root directory.
 
+import argparse
 import os
 
+from benchmarks.definition import Benchmark
 from benchmarks.domains import BENCHMARKS
 from benchmarks.persistence import (
     LocalFileObjectStorage,
     PathConstructor,
     S3ObjectStorage,
 )
+from benchmarks.result import Result
 
 RUNS_IN_CI = "CI" in os.environ
 
 
-def main():
+def save_benchmark_data(benchmark: Benchmark, result: Result):
```

Comment:
lets add return types
 ```suggestion
def save_benchmark_data(benchmark: Benchmark, result: Result) -> None:
```